pkg/sharding_engine/simple/sql_se.go

```diff
@@ -0,0 +1,75 @@
+package simple
+
+import (
+	"context"
+	"fmt"
+	pb "github.com/anujga/dstk/pkg/api/proto"
+	"github.com/jmoiron/sqlx"
+	"go.uber.org/zap"
+	"strings"
+	"time"
+)
+
+type sqlSe struct {
+	db *sqlx.DB
+}
+
+func (s *sqlSe) CreatePartition(ctx context.Context, request *pb.PartitionCreateRequest) (*pb.PartitionRpcBaseResponse, error) {
+	p := request.GetPartition()
+	p.ModifiedOn = time.Now().Unix()
+	q := `INSERT INTO partition 
+    		(id, modified_on, worker_id, start, "end", url, desired_state, current_state)
+    	 VALUES
+			(:id, :modified_on, :worker_id, :start, :end, :url, :desired_state, :current_state)
+	`
+	l := zap.L().With(zap.Any("part", p))
+	if result, err := s.db.NamedExec(q, *FromProto(p)); err == nil {
+		l.Info("partition created", zap.Any("result", result))
+		return &pb.PartitionRpcBaseResponse{}, nil
+	} else {
+		l.Error("partition creation failed", zap.Error(err))
+		return nil, err
```

Comment:
since this is a direct internet facing call, you need to wrap the erros inside a `grpc.status`